### PR TITLE
docs: optimize graceful_shutdown docs

### PR DESCRIPTION
**Pull request by zstone12**
#### What this PR does / why we need it (en: English/zh: Chinese):

en: optimize graceful_shutdown docs
zh: 完善优雅退出文档

#### Which issue(s) this PR fixes:

https://github.com/cloudwego/hertz/issues/415

### CODE REVIEW

File: content/zh/docs/hertz/tutorials/basic-feature/graceful-shutdown.md [hunk 1]
```diff
@@ -23,3 +23,17 @@ Hertz 支持优雅退出，优雅退出过程如下：
...
+
+Hertz 使用 `waitSignal` 作为信号处理的默认实现方式，处理方式如下:
+- 当接收到 `SIGTERM` 系统信号时触发立即退出。
+- 当接收到 `SIGHUP|SIGINT` 系统信号时触发优雅退出。
+
+当信号处理的默认实现方式无法满足需求时，可通过 `SetCustomSignalWaiter` 来进行自定义信号处理。
+```go
+  h := server.New()
+	h.SetCustomSignalWaiter(func(err chan error) error {
+		return nil
```
Comment (li-jin-gou) [inline content/zh/docs/hertz/tutorials/basic-feature/graceful-shutdown.md:35]: This example code does not appear to be instructive.

**Review by li-jin-gou — CHANGES_REQUESTED** [review]

File: content/zh/docs/hertz/tutorials/basic-feature/graceful-shutdown.md (hunk 1 above)
Comment (zstone12) [reply content/zh/docs/hertz/tutorials/basic-feature/graceful-shutdown.md:35]: got it. tks

File: content/en/docs/hertz/tutorials/basic-feature/graceful-shutdown.md [hunk 2]
```diff
@@ -23,3 +23,16 @@ If you want to register the `hook` function, you can do so by getting the `Engin
 ```go
 h.Engine.OnShutdown = append(h.Engine.OnShutdown, shutDownFunc)
 ```
+
```
Comment (li-jin-gou) [inline content/en/docs/hertz/tutorials/basic-feature/graceful-shutdown.md:26]: Does the way in which the maximum waiting time is adjusted need to be specified?

**Review by li-jin-gou — APPROVED** [review]

**Comment by li-jin-gou** [comment]
LGTM

**Pull request merged**